### PR TITLE
Separate arm64 and amd64 docker builds

### LINKED ISSUE #125094: Broken Docker Image on dockerhub
### 🐛 Describe the bug

It appears that the docker image on [dockerhub](https://hub.docker.com/r/pytorch/pytorch/tags) for 2.3.0 cuda11.8 & cuda12.1, both runtime and devel are all malformed. 
1. They target arm64 instead of amd64 like all previous images
2. The copy from `/opt/conda` only copied ~300Mb where it normaly copies several Gb. 

### Versions

N/A. Image will not run. 

cc @ezyang @gchanan @zou3519 @kadeng

### PR DESCRIPTION
**Pull request by atalman**
Fixes https://github.com/pytorch/pytorch/issues/125094

Please note: Docker CUDa 12.4 failure is existing issue, related to docker image not being available on gitlab: 
```
docker.io/nvidia/cuda:12.4.0-cudnn8-devel-ubuntu22.04: docker.io/nvidia/cuda:12.4.0-cudnn8-devel-ubuntu22.04: not found
```
 https://github.com/pytorch/pytorch/actions/runs/8974959068/job/24648540236?pr=125617

Here is the reference issue: https://gitlab.com/nvidia/container-images/cuda/-/issues/225

Tracked on our side: https://github.com/pytorch/builder/issues/1811

### CODE REVIEW

File: .github/workflows/docker-release.yml [hunk 1]
```diff
@@ -128,18 +129,29 @@ jobs:
...
 
           docker tag ghcr.io/pytorch/pytorch-nightly:"${PYTORCH_DOCKER_TAG}" \
-                 ghcr.io/pytorch/pytorch-nightly:"${PYTORCH_NIGHTLY_COMMIT}-cu${CUDA_VERSION}"
-          docker push ghcr.io/pytorch/pytorch-nightly:"${PYTORCH_NIGHTLY_COMMIT}-cu${CUDA_VERSION}"
+                 ghcr.io/pytorch/pytorch-nightly:"${PYTORCH_NIGHTLY_COMMIT}${CUDA_SUFFIX}"
+
+          docker push ghcr.io/pytorch/pytorch-nightly:"${PYTORCH_NIGHTLY_COMMIT}${CUDA_SUFFIX}"
+
+          # Please note, here we ned to pin specific verison of CUDA as with latest label
+          if[[${CUDA_VERSION_SHORT} == "12.1"]]; then
```
Comment (atalman) [inline .github/workflows/docker-release.yml:149]: Do we need latest label at all here ? Maybe we can simply remove it. I see following stats:
https://github.com/orgs/pytorch/packages/container/pytorch-nightly/212425390?tag=latest
```
Download activity
Total downloads
0
Last 30 days
0
Last week
0
Today
0

```

**Review by huydhn — APPROVED** [review]
Plz fix lint before landing

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**